### ISSUE TITLE
Printing corrupted images

### ISSUE THREAD

**Issue by pkronstrom** [BoboTiG/thermalprinter#9]
Greetings! Thank you for this awesome library!

I've been wondering for a while the reason why I occasionally get corrupted images (missing rows, or rows not aligned correctly). Sometimes the image printing gets halted and the printer starts spewing out random characters.

I looked into your code and noticed that the image printing script sleeps for a short amount of time for each chunk's row. This is dependant on the printer's baudrate.

See https://github.com/BoboTiG/thermalprinter/blob/master/thermalprinter/thermalprinter.py#L418 and
https://github.com/BoboTiG/thermalprinter/blob/master/thermalprinter/thermalprinter.py#L53

I managed to get my images working by setting a hardcoded self._byte_Time 0.0005 (which is somewhat close to the default timing) rather than depending on the low sleep time set by my higher baudrate 115200.

Just to let you know! Thanks again!

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/tiger-222) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/BoboTiG/thermalprinter/issues/9">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/BoboTiG/thermalprinter/issues/9/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/BoboTiG/thermalprinter/issues/9/pledge.svg">
</picture>
</a>

**Comment by aleray** [comment]
Hello,

I'm having the exact same issue, with any images.

I send 384px wide pictures, in 1bit. Usually, it starts with missing/miss-aligned rows and then I get some kind of buffer overflow with junk characters printed.

My printer is at the default 19200 baud rate. It is a [cheap Ada Fruit Mini Thermal Printer](https://cdn-learn.adafruit.com/downloads/pdf/mini-thermal-receipt-printer.pdf?timestamp=1551161979).

Thanks

**Comment by BoboTiG** [comment]
@aleray could you post your code and the picture you are trying to print please?

**Comment by aleray** [comment]
Hello @BoboTiG here is my code. Image to follow...

```python
import sys
from PIL import Image
from thermalprinter import *

with ThermalPrinter(port='/dev/ttyUSB0', most_heated_point=6) as printer:
    for img in sys.argv[1:]:
        # Picture
        printer.image(Image.open(img))
```

**Comment by BoboTiG** [comment]
Thanks, I will have a look when possible (likely this week-end).

**Comment by aleray** [comment]
@BoboTiG Thanks.

In the meanwhile I tested <https://github.com/adafruit/Python-Thermal-Printer> and it seems to be working properly, although the code is not well packaged and it might be Python 2 only.

Have a look at their source code, they extensively commented it and they talk about a few trick to avoid buffer overflows.

**Comment by sharkwheels** [comment]
I'm also getting this issue. Sometimes its the buffer overflow, sometimes its just missing lines in the image. Any update on this?

**Comment by BoboTiG** [comment]
I will have a look this week, sorry for the inconvenience :)

**Comment by BoboTiG** [comment]
OK I tried something but without success. I will check later (I need to find time).
If anyone wants to give a hand, help is very welcome :)

**Comment by BoboTiG** [comment]
I released the version [1.0.0](https://github.com/BoboTiG/thermalprinter/releases/tag/1.0.0) with a new code to print images. I close the issue since it's an old one. Just open a new issue on any other problem.
